src/Microsoft.AspNet.Hosting/WebApplicationBuilder.cs

```diff
@@ -228,7 +167,7 @@ public WebHostBuilder UseStartup(Action<IApplicationBuilder> configureApp, Func<
             return this;
         }
 
-        public WebHostBuilder UseStartup(Action<IApplicationBuilder> configureApp, Action<IServiceCollection> configureServices)
+        public WebApplicationBuilder UseStartup(Action<IApplicationBuilder> configureApp, Action<IServiceCollection> configureServices)
```

Comment:
Note we still need to get rid of the conflicting overload here between `Action<IServiceCollection>` and `Func<IServiceCollection, IServiceProvider>`